packages/coding-style/src/Rector/ClassMethod/AnnotateThrowablesRector.php

```diff
@@ -0,0 +1,174 @@
+                     // code before
+                    <<<'PHP'
+class RootExceptionInMethodWithDocblock
+{
+    /**
+     * This is a comment.
+     *
+     * @param int $code
+     */
+    public function throwException(int $code)
+    {
+        throw new \RuntimeException('', $code);
+    }
+}
+PHP
+                    ,
+                    // code after
+                    <<<'PHP'
+class RootExceptionInMethodWithDocblock
+{
+    /**
+     * This is a comment.
+     *
+     * @param int $code
+     * @throws \RuntimeException
+     */
+    public function throwException(int $code)
+    {
+        throw new \RuntimeException('', $code);
+    }
+}
+PHP
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @param Throw_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->isThrowableAnnotated($node)) {
+            return null;
+        }
+
+        $this->annotateMethod($node);
+
+        return $node;
+    }
+
+    private function isThrowableAnnotated(Throw_ $node): bool
+    {
+        $method = $node->getAttribute(AttributeKey::METHOD_NODE);
+
+        if ($method === null) {
+            throw new RuntimeException('This should not happen and is probably a bug. Please report it.');
+        }
+
+        /** @var PhpDocInfo $phpDocInfo */
+        $phpDocInfo = $method->getAttribute(AttributeKey::PHP_DOC_INFO);
+        $throwTags = $phpDocInfo->getTagsByName('throws');
+        $FQN = $this->buildFQN($node);
+
+        if (empty($throwTags)) {
+            return false;
+        }
+
+        /** @var AttributeAwarePhpDocTagNode $throwTag */
+        foreach ($throwTags as $throwTag) {
+            $throwClassName = $throwTag->value->type->name;
+            if ($throwClassName === $FQN) {
+                return true;
+            }
+
+            if (
+                ! Strings::contains($throwClassName, '\\') &&
+                Strings::contains($FQN, $throwClassName) &&
+                $this->isThrowableImported($node)
+            ) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private function isThrowableImported(Throw_ $node): bool
+    {
+        $throwClassName = $this->getName($node->expr->class);
+        $useNodes = $node->getAttribute(AttributeKey::USE_NODES);
+
+        if ($useNodes === null) {
+            return false;
+        }
+
+        /** @var Use_ $useNode */
+        foreach ($useNodes as $useNode) {
+            /** @var UseUse $useStmt */
+            foreach ($useNode->uses as $useStmt) {
+                if ($this->getName($useStmt) === $throwClassName) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private function annotateMethod(Throw_ $node): void
+    {
+        /** @var ClassMethod $method */
+        $method = $node->getAttribute(AttributeKey::METHOD_NODE);
+        $FQN = $this->buildFQN($node);
```

Comment:
This `$FQN` was already resolved before. It shold be in argument of this method.

Also FQN is too vague name. It could be any class. But actually this is `$exceptionClass` or `$throwableClass`, maybe use that instead